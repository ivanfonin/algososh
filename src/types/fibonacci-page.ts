export type TNumber = {
  letter: string;
  index: number;
};
