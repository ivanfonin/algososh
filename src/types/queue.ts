import { ElementStates } from "./element-states";

export type TQueueItem = {
  state: ElementStates;
  letter: string;
};
