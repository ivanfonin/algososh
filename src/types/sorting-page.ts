import { ElementStates } from "./element-states";

export enum Algorithms {
  Select = "select",
  Bubble = "bubble",
}

export type TColumn = {
  index: number;
  state: ElementStates;
};
