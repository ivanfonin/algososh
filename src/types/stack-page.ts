import { ElementStates } from "./element-states";

export type TStackItem = {
  state: ElementStates;
  letter: string;
};
