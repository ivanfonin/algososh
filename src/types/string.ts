import { ElementStates } from "./element-states";

export type TLetter = {
  state: ElementStates;
  letter: string;
};
