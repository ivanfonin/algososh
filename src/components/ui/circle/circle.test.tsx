import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

const circleProps = {
  state: ElementStates.Default,
  letter: "test",
  head: "str",
  index: 1,
  tail: "str",
  extraClass: "extra-class",
  isSmall: true,
};

describe("Тестируем компонент Circle", () => {
  afterEach(cleanup);

  it("Кружок с пропсами рендерится без ошибок", () => {
    const tree = renderer.create(<Circle {...circleProps} />);
    expect(tree).toMatchSnapshot();
  });

  it("Кружок без буквы рендерится без ошибок", () => {
    const tree = renderer.create(<Circle />);
    expect(tree).toMatchSnapshot();
  });

  it("Кружок с буквами рендерится без ошибок", () => {
    const tree = renderer.create(<Circle letter="Abc" />);
    expect(tree).toMatchSnapshot();
  });

  it("Кружок с head рендерится без ошибок", () => {
    const tree = renderer.create(<Circle head="Abc" />);
    expect(tree).toMatchSnapshot();
  });

  it("Кружок с react-элементом в head рендерится без ошибок", () => {
    const tree = renderer.create(<Circle head={<Circle letter="head" />} />);
    expect(tree).toMatchSnapshot();
  });

  it("Кружок с tail рендерится без ошибок", () => {
    const tree = renderer.create(<Circle tail="Abc" />);
    expect(tree).toMatchSnapshot();
  });

  it("Кружок с react-элементом в tail рендерится без ошибок", () => {
    const tree = renderer.create(<Circle tail={<Circle letter="tail" />} />);
    expect(tree).toMatchSnapshot();
  });

  it("Кружок с пропсом isSmall === true рендерится без ошибок", () => {
    const tree = renderer.create(<Circle isSmall={true} />);
    expect(tree).toMatchSnapshot();
  });

  it("Кружок в состоянии default рендерится без ошибок", () => {
    const tree = renderer.create(<Circle state={ElementStates.Default} />);
    expect(tree).toMatchSnapshot();
  });

  it("Кружок в состоянии changing рендерится без ошибок", () => {
    const tree = renderer.create(<Circle state={ElementStates.Changing} />);
    expect(tree).toMatchSnapshot();
  });

  it("Кружок в состоянии modified рендерится без ошибок", () => {
    const tree = renderer.create(<Circle state={ElementStates.Modified} />);
    expect(tree).toMatchSnapshot();
  });
});
