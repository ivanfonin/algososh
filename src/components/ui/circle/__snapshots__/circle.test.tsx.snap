// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестируем компонент Circle Кружок без буквы рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Circle Кружок в состоянии changing рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Circle Кружок в состоянии default рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Circle Кружок в состоянии modified рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Circle Кружок с head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    Abc
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Circle Кружок с react-элементом в head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          head
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Circle Кружок с react-элементом в tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          tail
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тестируем компонент Circle Кружок с tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    Abc
  </div>
</div>
`;

exports[`Тестируем компонент Circle Кружок с буквами рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестируем компонент Circle Кружок с пропсами рендерится без ошибок 1`] = `
<div
  className="content extra-class"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    str
  </div>
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      test
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 string"
  >
    str
  </div>
</div>
`;

exports[`Тестируем компонент Circle Кружок с пропсом isSmall === true рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
